runtime/lua/vim/treesitter/_fold.lua

```diff
@@ -147,6 +147,22 @@ local function normalise_erow(bufnr, erow)
   return math.min(erow or max_erow, max_erow)
 end
 
+---@type table<string, Query|false|nil>
+--- `false` if query was not found
+local queries = {}
+
+---@return Query|nil
+local function get_query(lang)
```

Comment:
This looks like [another case](https://github.com/neovim/neovim/blob/ff8e3c6039d50bdbc5b19ac468b77567cb79d04a/runtime/lua/vim/diagnostic.lua#L387) that could benefit from a generalized "memoize" / LRU mechanism.  #23233

Can you add a comment here like:

```lua
-- TODO: use generalized memoize / LRU. #23233
```

Reply:
There's a memoize function in nvim-treesitter (main branch); that could be upstreamed.